tests/unit/blocks/blockRegistration.test.js

```diff
@@ -0,0 +1,174 @@
+ * @param {object} builtProps Props built for this object.
+ */
+const recursiveAssertProps = (expectedProps, builtProps) => {
+
+	// Some sanity checkcs
+	expect(
+		builtProps,
+		`Missing attributes key in built props for: ${debugName(builtProps)}`
+	).toHaveProperty('attributes');
+
+	// Check expected
+	for (const expected of expectedProps.expected) {
+		expect(
+			builtProps.attributes,
+			`Missing expected property in built props for: ${debugName(builtProps)}`
+		).toHaveProperty(expected);
+	}
+
+	// Check not expected
+	for (const notExpected of expectedProps.notExpected) {
+		expect(builtProps.attributes).not.toHaveProperty(notExpected);
+	}
+
+	// Check prefix
+	expect(
+		builtProps.attributes,
+		`Missing "prefix" key in built props for ${debugName(builtProps)}`
+	).toHaveProperty('prefix');
+	expect(
+		builtProps.attributes.prefix,
+		`Prefix key not correct when sending props to ${debugName(builtProps)}`
+	).toEqual(expectedProps.prefix);
+
+	// Repeat if expected has subComponents
+	if (expectedProps.components && Object.keys(expectedProps.components).length > 0) {
+		for (const componentKey of Object.keys(expectedProps.components)) {
+
+			// Expect built props has subComponents
+			expect(
+				buildProps,
+				`Missing "subComponents" key in built props for ${debugName(builtProps)}. Expecting one because this has subComponents defined in props-output.js`
+			).toHaveProperty('subComponents');
+			expect(
+				buildProps.subComponents,
+				`"subComponents" key in built props empty for ${debugName(builtProps)}.`
+			).not.toEqual([]);
+
+			// Find the correct subComponent
+			const subComponent = builtProps.subComponents.filter((subComponent) => subComponent.newName === componentKey);
+
+			// Expect to find exactly 1 sub component
+			expect(
+				subComponent,
+				`Expected subComponent not found in built props for ${debugName(builtProps)}.`
+			).toHaveLength(1);
+
+			// Repeat for all subComponents
+			recursiveAssertProps(expectedProps.components[componentKey], subComponent[0]);
+		}
+	}
+}
+
+/**
+ * Unit tests for get-option-colors.js helper
+ *
+ * @group unit
+ */
+it('tests block registration properly inherits all component attributes', () => {
+
+	const componentManifests = getAllComponentManifests();
+	const blockManifests = getAllBlockManifests();
+
+	// Build an array of expected attributes in each block.
+	buildWindowObject(globalManifest, componentManifests, blockManifests, wrapperManifest);
+
+	// Test all block manifests.
+	for (const blockManifest of blockManifests) {
+		const blockName = blockManifest.blockName;
+		const attributeNames = Object.keys(getAttributes(globalManifest, wrapperManifest, componentManifests, blockManifest));
+
+		// Only test blocks which expected / not expected attributes defined.
+		if (!blockAttributes[blockName]) {
+			continue;
+		}
+
+		// Make sure the block contains all expected attributes
+		if (blockAttributes[blockName].expected) {
+			expect(
```

Comment:
Extra enter here?

Reply:
Not sure what you mean? This has the attributes spread over 2 lines so the message is more visible.

Reply:
Looked a bit odd for me on the first look, that's why I asked. Maybe should've phrased it better 😄  It's all good now.